src/main/java/com/google/api/generator/gapic/composer/rest/RestContext.java

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.gapic.composer.rest;
+
+import com.google.api.gax.core.BackgroundResource;
+import com.google.api.gax.httpjson.ApiMethodDescriptor;
+import com.google.api.gax.httpjson.HttpJsonCallSettings;
+import com.google.api.gax.httpjson.HttpJsonCallableFactory;
+import com.google.api.gax.httpjson.HttpJsonStubCallableFactory;
+import com.google.api.gax.httpjson.HttpJsonTransportChannel;
+import com.google.api.gax.httpjson.InstantiatingHttpJsonChannelProvider;
+import com.google.api.generator.gapic.composer.common.TransportContext;
+import com.google.api.generator.gapic.composer.utils.ClassNames;
+import com.google.api.generator.gapic.model.Transport;
+
+public abstract class RestContext extends TransportContext {
+  private static final TransportContext INSTANCE =
+      RestContext.builder()
+          .setClassNames(new ClassNames("HttpJson"))
+          .setTransport(Transport.REST)
+          .setTransportName("REST")
+          // For httpjson.HttpJsonServiceStubClassComposer
+          .setCallSettingsClass(HttpJsonCallSettings.class)
+          .setStubCallableFactoryType(classToType(HttpJsonStubCallableFactory.class))
+          .setMethodDescriptorClass(ApiMethodDescriptor.class)
+          .setTransportOperationsStubType(null)
+          // For httpjson.ServiceSettingsClassComposer
+          .setInstantiatingChannelProviderClass(InstantiatingHttpJsonChannelProvider.Builder.class)
+          .setDefaultTransportProviderBuilderName("defaultHttpJsonTransportProviderBuilder")
+          // For httpjson.ServiceStubSettingsClassComposer
+          .setTransportChannelType(classToType(HttpJsonTransportChannel.class))
+          .setTransportGetterName("getHttpJsonTransportName")
+          // For httpjson.HttpJsonServiceCallableFactoryClassComposer
+          .setTransportCallSettingsType(classToType(HttpJsonCallSettings.class))
+          .setTransportCallableFactoryType(classToType(HttpJsonCallableFactory.class))
+          .setOperationsStubType(classToType(BackgroundResource.class))
+          .setTransportCallSettingsName("httpJsonCallSettings")
+          .build();
+
+  public static TransportContext instance() {
```

Comment:
Methods usually cause less troubles than direct usage of the fields in tests. That is more theory, though, if the stuff is static, it is hard to use that in test either way (sometimes almost impossible, without nasty reflection/reflection-based mocking libraries or mocking way too much). 

But most importantly, I was basically following the same style, which has been used for all other singleton classes in this repository, composer classes in particular. Also, accessing the singleton class instance via a satatic method instead of a static variable is a part of the singleton pattern itself (for example https://en.wikipedia.org/wiki/Singleton_pattern, check examples section). In general, method always gives more flexibility, in case of singletons, for example it gives a flexibility of lazy vs eager initialization strategy.

Reply:
SGTM